scrapy_poet/utils/testing.py

```diff
@@ -229,26 +232,46 @@ def process_response(self, request, response, spider):
         return response
 
 
-def create_scrapy_settings():
-    """Default scrapy-poet settings"""
-    s = dict(
+def _get_test_settings():
+    settings = {
         # collect scraped items to crawler.spider.collected_items
-        ITEM_PIPELINES={
+        "ITEM_PIPELINES": {
             CollectorPipeline: 100,
         },
-        DOWNLOADER_MIDDLEWARES={
+        "DOWNLOADER_MIDDLEWARES": {
             # collect injected dependencies to crawler.spider.collected_response_deps
             InjectedDependenciesCollectorMiddleware: 542,
-            "scrapy_poet.InjectionMiddleware": 543,
-            "scrapy.downloadermiddlewares.stats.DownloaderStats": None,
-            "scrapy_poet.DownloaderStatsMiddleware": 850,
         },
-        REQUEST_FINGERPRINTER_CLASS=ScrapyPoetRequestFingerprinter,
-        SPIDER_MIDDLEWARES={
+    }
+    try:
+        import scrapy.addons  # noqa: F401
+    except ImportError:
+        settings["DOWNLOADER_MIDDLEWARES"]["scrapy_poet.InjectionMiddleware"] = 543
+        settings["DOWNLOADER_MIDDLEWARES"][
+            "scrapy.downloadermiddlewares.stats.DownloaderStats"
+        ] = None
+        settings["DOWNLOADER_MIDDLEWARES"][
+            "scrapy_poet.DownloaderStatsMiddleware"
+        ] = 850
+        settings["REQUEST_FINGERPRINTER_CLASS"] = ScrapyPoetRequestFingerprinter
+        settings["SPIDER_MIDDLEWARES"] = {
             "scrapy_poet.RetryMiddleware": 275,
-        },
+        }
+    else:
+        settings["ADDONS"] = {
+            "scrapy_poet.Addon": 300,
+        }
+    return settings
+
+
+def create_scrapy_settings():
+    """Return the default scrapy-poet settings."""
+    warn(
+        "The scrapy_poet.utils.create_scrapy_settings() function is deprecated.",
+        DeprecationWarning,
+        stacklevel=2,
     )
-    return Settings(s)
+    return Settings(_get_test_settings())
```

Comment:
Deprecated because the returned settings include all Scrapy default settings as project-level settings, which prevents the add-on-level request fingerprinting from being taken into account.